packages/next/next-server/server/render.tsx

```diff
@@ -307,7 +307,7 @@ const invalidKeysMsg = (methodName: string, invalidKeys: string[]) => {
 }
 
 type Redirect = {
-  permanent: boolean
+  permanent?: boolean
   destination: string
   statusCode?: number
```

Comment:
Why is this type duplicated instead of using the same or type from `packages/next/types/index.d.ts`? Could you please dedupe this?

Reply:
I don't know why the type is duplicated because I didn't write this. @ijjk wrote in https://github.com/vercel/next.js/commit/bc80fb4fb2530c9730ea6d2c7e118d31ec7965f4.

I think the type here should not be so strict because `checkRedirectValues` below has to validate JavaScript inputs, while the type from `packages/next/types/index.d.ts` should be strict for TypeScript users.